### ISSUE TITLE
An URL in parsed-literal contents gets wrongly rendered in PDF if with hyphen

### ISSUE THREAD

**Issue by jfbu** [sphinx-doc/sphinx#3317]
Subject: An URL in parsed-literal contents gets wrongly rendered in PDF if with hyphen

#### Procedure to reproduce the problem

```rst
.. parsed-literal::

   http://www.sphinx-doc.org

   *incite Sphinx to recognize this as parsed-literal*

```

#### Error logs / results

![capture d ecran 2017-01-08 a 15 33 52](https://cloud.githubusercontent.com/assets/2589111/21750633/57f03c00-d5b8-11e6-9418-748dc8a1371d.png)

#### Expected results

The `-` does not get converted to obscure LaTeX code... 

### Reproducible project / your project

[urlinparsedliteral.zip](https://github.com/sphinx-doc/sphinx/files/692163/urlinparsedliteral.zip)

### Environment info
- OS: Mac
- Python version: 3.5
- Sphinx version: 1.5.1

**Comment by jfbu** [comment]
The produced latex contains `\url{http://www.sphinx-doc.org}`, but the alltt environment has made the `-` active. It appears that `\url` inside the alltt has the effect of replacing `-` with a verbatim rendering of the latex meaning of the active `-`.

**Comment by jfbu** [comment]
closed via PR #3340 at 5e0a764
